packages/wagmi/src/index.ts

```diff
@@ -23,11 +27,31 @@ export function bindings(): IBindings {
         }
       }
 
-      const signer = await fetchSigner<RequiredSigner>({ chainId });
+      const walletClient = await getWalletClient({ chainId });
 
-      invariant(signer, 'Cannot get signer, is the wallet connected?');
+      invariant(walletClient, 'Wallet client not found');
 
-      return signer;
+      return signerFromWalletClient({ walletClient, chainId });
     },
   };
 }
+
+export function providerFromPublicClient({
+  publicClient,
+  chainId,
+}: {
+  publicClient: PublicClient;
+  chainId?: number;
```

Comment:
nit:, you could get the `chainId` from the `publicClient.chain`, same with `signerFromWalletClient`

Reply:
tried this and it sometimes came back as undefined?